src/crystal/system/win32/file_descriptor.cr

```diff
@@ -519,7 +519,11 @@ private module ConsoleUtils
   @@read_requests = Deque(ReadRequest).new
   @@bytes_read = Deque(Int32).new
   @@mtx = ::Thread::Mutex.new
-  @@reader_thread = ::Thread.new { reader_loop }
+  {% if flag?(:execution_context) %}
+    @@reader_thread = ::Fiber::ExecutionContext.new("READER-LOOP") { reader_loop }
```

Comment:
```suggestion
    @@reader_thread = ::Fiber::ExecutionContext::Isolated.new("READER-LOOP") { reader_loop }
```

Reply:
:facepalm:

Reply:
Considering this, it would be nice to have at least some basic smoke testing for this code. I suppose requires #15511 first though.